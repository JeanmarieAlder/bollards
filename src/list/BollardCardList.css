.bollard-card-list-div{
    text-align: center;
}

.search-result-title {
    font-family: 'Kalam';
    font-weight: 700;
    text-align: center;
}