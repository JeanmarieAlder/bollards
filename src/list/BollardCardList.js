import React from 'react';
import BollardCard from './BollardCard';

import './BollardCardList.css';

const BollardCardList = ({bollards}) => {
    return (
        <div className='p-5 card-container row text-center justify-content-center'>
            {
                bollards.map((bollard, i) => {
                    return (
                        <BollardCard key={bollards[i].id} bollard={bollards[i]} />
                    );
                } )
            }
        </div>
    );
}

export default BollardCardList;