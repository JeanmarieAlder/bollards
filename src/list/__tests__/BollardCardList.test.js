import React from "react";
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import BollardCardList from "../BollardCardList";
import { test_bollards_list } from "../../tests/TestData";
import { MemoryRouter } from "react-router";

test("Component displays correctly", () => {
    const component = render(
        <MemoryRouter>
            <BollardCardList bollards={test_bollards_list}></BollardCardList>
        </MemoryRouter>
    );
    let tree = component.asFragment();
    console.log(tree)
    expect(tree).toMatchSnapshot();
})


test("persistScrollPosition method calls sessionStorage", () => {
    window.sessionStorage.setItem = jest.fn(() => Promise.resolve())
    const shallowRenderer = new ShallowRenderer();
    const component = shallowRenderer.render(
        <BollardCardList bollards={test_bollards_list}></BollardCardList>
    );
    console.log(component);
    const result = shallowRenderer.getRenderOutput();
    console.log(result);
    const child = result.props.children[0]
    console.log(child);
    child.props.onSelectBollard(42);
    expect(window.sessionStorage.getItem('scroll-position-bollard-id')).toEqual("42");
})