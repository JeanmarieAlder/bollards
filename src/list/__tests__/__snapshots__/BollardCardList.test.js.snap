// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component displays correctly 1`] = `
<DocumentFragment>
  <div
    class="p-2 card-container row text-center justify-content-center"
  >
    <div
      class="col-6 col-md-3 bollard-card"
      id="7"
    >
      <a
        href="/details/8"
        style="text-decoration: none;"
      >
        <div
          class="card"
        >
          <div
            class="img-loading-div"
          >
            <div
              class="spinner-grow text-secondary"
              role="status"
            />
          </div>
          <div>
            <img
              alt="bollard"
              class="img-preview"
              src="http://localhost:5000/static/img/bollards_icon/b9c559e2cdd6e8d3.jpg"
            />
          </div>
          <div
            class="card-body bollard-card-text-div"
          >
            <h2
              class="card-title"
            >
              No 7
            </h2>
            <p
              class="card-text"
            >
              <i />
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      class="col-6 col-md-3 bollard-card"
      id="10"
    >
      <a
        href="/details/11"
        style="text-decoration: none;"
      >
        <div
          class="card"
        >
          <div
            class="img-loading-div"
          >
            <div
              class="spinner-grow text-secondary"
              role="status"
            />
          </div>
          <div>
            <img
              alt="bollard"
              class="img-preview"
              src="http://localhost:5000/static/img/bollards_icon/b673f71ca97afdcf.jpg"
            />
          </div>
          <div
            class="card-body bollard-card-text-div"
          >
            <h2
              class="card-title"
            >
              No 10
            </h2>
            <p
              class="card-text"
            >
              <i>
                Beuffarde
              </i>
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      class="col-6 col-md-3 bollard-card"
      id="11"
    >
      <a
        href="/details/15"
        style="text-decoration: none;"
      >
        <div
          class="card"
        >
          <div
            class="img-loading-div"
          >
            <div
              class="spinner-grow text-secondary"
              role="status"
            />
          </div>
          <div>
            <img
              alt="bollard"
              class="img-preview"
              src="http://localhost:5000/static/img/bollards_icon/d9093d663fb67db6.jpg"
            />
          </div>
          <div
            class="card-body bollard-card-text-div"
          >
            <h2
              class="card-title"
            >
              No 11
            </h2>
            <p
              class="card-text"
            >
              <i>
                Coin de la Beuffarde
              </i>
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      class="col-6 col-md-3 bollard-card"
      id="27"
    >
      <a
        href="/details/30"
        style="text-decoration: none;"
      >
        <div
          class="card"
        >
          <div
            class="img-loading-div"
          >
            <div
              class="spinner-grow text-secondary"
              role="status"
            />
          </div>
          <div>
            <img
              alt="bollard"
              class="img-preview"
              src="http://localhost:5000/static/img/bollards_icon/1b5e007272220edb.jpg"
            />
          </div>
          <div
            class="card-body bollard-card-text-div"
          >
            <h2
              class="card-title"
            >
              No 27
            </h2>
            <p
              class="card-text"
            >
              <i />
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      class="col-6 col-md-3 bollard-card"
      id="28"
    >
      <a
        href="/details/31"
        style="text-decoration: none;"
      >
        <div
          class="card"
        >
          <div
            class="img-loading-div"
          >
            <div
              class="spinner-grow text-secondary"
              role="status"
            />
          </div>
          <div>
            <img
              alt="bollard"
              class="img-preview"
              src="http://localhost:5000/static/img/bollards_icon/91daf6e9e0dc99fd.jpg"
            />
          </div>
          <div
            class="card-body bollard-card-text-div"
          >
            <h2
              class="card-title"
            >
              No 28
            </h2>
            <p
              class="card-text"
            >
              <i />
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      class="col-6 col-md-3 bollard-card"
      id="33"
    >
      <a
        href="/details/36"
        style="text-decoration: none;"
      >
        <div
          class="card"
        >
          <div
            class="img-loading-div"
          >
            <div
              class="spinner-grow text-secondary"
              role="status"
            />
          </div>
          <div>
            <img
              alt="bollard"
              class="img-preview"
              src="http://localhost:5000/static/img/bollards_icon/84bcd953b6cb9785.jpg"
            />
          </div>
          <div
            class="card-body bollard-card-text-div"
          >
            <h2
              class="card-title"
            >
              No 33
            </h2>
            <p
              class="card-text"
            >
              <i />
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      class="col-6 col-md-3 bollard-card"
      id="36"
    >
      <a
        href="/details/39"
        style="text-decoration: none;"
      >
        <div
          class="card"
        >
          <div
            class="img-loading-div"
          >
            <div
              class="spinner-grow text-secondary"
              role="status"
            />
          </div>
          <div>
            <img
              alt="bollard"
              class="img-preview"
              src="http://localhost:5000/static/img/bollards_icon/6b6f318eb9701310.jpg"
            />
          </div>
          <div
            class="card-body bollard-card-text-div"
          >
            <h2
              class="card-title"
            >
              No 36
            </h2>
            <p
              class="card-text"
            >
              <i />
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      class="col-6 col-md-3 bollard-card"
      id="51"
    >
      <a
        href="/details/54"
        style="text-decoration: none;"
      >
        <div
          class="card"
        >
          <div
            class="img-loading-div"
          >
            <div
              class="spinner-grow text-secondary"
              role="status"
            />
          </div>
          <div>
            <img
              alt="bollard"
              class="img-preview"
              src="http://localhost:5000/static/img/bollards_icon/90ed7af297c625fb.jpg"
            />
          </div>
          <div
            class="card-body bollard-card-text-div"
          >
            <h2
              class="card-title"
            >
              No 51
            </h2>
            <p
              class="card-text"
            >
              <i />
            </p>
          </div>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
